rust/kernel/module_param.rs

```diff
@@ -27,7 +34,13 @@ pub trait ModuleParam: core::fmt::Display + core::marker::Sized {
     /// `arg == None` indicates that the parameter was passed without an
     /// argument. If `NOARG_ALLOWED` is set to `false` then `arg` is guaranteed
     /// to always be `Some(_)`.
-    fn try_from_param_arg(arg: Option<&[u8]>) -> Option<Self>;
+    fn try_from_param_arg(arg: Option<&'static [u8]>) -> Option<Self>;
```

Comment:
Why is it now `'static`?

Reply:
This allows the parameter to store a reference to the argument as we're doing for the string parameter. I think it's safe to treat it as `'static` since `Self` will be updated whenever the arg is changed (and obviously can't be used after the module is unloaded).

One alternative would be to have `Self` take a lifetime parameter, but then all the types that never keep references to the argument will have an extra ignored parameter.

Maybe another alternative would be to leave the signature as-is here and do some more unsafe stuff in the `StringParam` implementation to convert the lifetime.

Could also be another possibility that I'm missing...

Reply:
Hopefully tomorrow I can give it a better look at all the `Ref` vs. `Owned` bits. @alex perhaps has more time.

By the way, we only support for the moment `initcall6`, which means allocations should be available in general, I think.

Reply:
Oh interesting. If that's the case we could have both the string and array types always allocate. That would simplify some things and also allow the array to be arbitrary length.

Reply:
Looked into this some more. `kmalloc` is available when the when `init` runs, but the parameters are actually set earlier than that and allocations don't work. So I think we should keep the current setup.